torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1253,27 +1278,35 @@ def _get_key(self):
         """
         return (len(self.replacements), len(self.divisible))
 
-    def _produce_dyn_sizes(self, ex: torch.Tensor, source: Source) -> List[sympy.Expr]:
+    def _produce_dyn_sizes(self,
+                           ex: torch.Tensor,
+                           source: Source,
+                           dims: Optional[List[DIM_DYNAMISM_STATE]]) -> List[sympy.Expr]:
         from torch._dynamo.source import TensorPropertySource, TensorProperty
         size = []
         for i, val in enumerate(ex.size()):
-            is_dynamic = _is_dim_dynamic(ex, i)
-            if _should_allocate(is_dynamic, self.assume_static_by_default):
+            dim_state = dims[i] if dims else DIM_DYNAMISM_STATE.DUCK
+            is_static = dim_state == DIM_DYNAMISM_STATE.STATIC
+            if not is_static:
                 size.append(self.create_symbol(
-                    val, TensorPropertySource(source, TensorProperty.SIZE, i), is_dynamic
+                    val, TensorPropertySource(source, TensorProperty.SIZE, i), dim_state
```

Comment:
Wouldn't it be more uniform to just have `create_symbol` return an int if dim_state is STATIC? (Rather than testing it here.)

Reply:
Yes, but this is the current logic, and I figured out of scope for this change.

Reply:
The smell here is you are doing equality tests against enums. What if you get more enum entries in the future? You have no way of easily finding all sites you need to update (whereas if you have an if-else switch on the enum, it's clear you have to add new handling for it.)

Reply:
sg